tutorial/lesson_07_multi_stage_pipelines.cpp

```diff
@@ -30,7 +31,7 @@ int main(int argc, char **argv) {
     // first horizontally, and then vertically.
     {
         // Take a color 8-bit input
-        Image<uint8_t> input = load<uint8_t>("images/rgb.png");
+        Image<uint8_t> input = load<Image<uint8_t>>("images/rgb.png");
```

Comment:
Yeah, a templated cast operator is fishy for sure, but I call this function
live on stage for tutorials, so I really want it to not confuse the
audience even for a second.

How about throwing an exception instead of exiting if WITH_EXCEPTIONS is on?

On Fri, Jul 31, 2015 at 10:43 AM, Steven Johnson <notifications@github.com>
wrote:

> In tutorial/lesson_07_multi_stage_pipelines.cpp
> <https://github.com/halide/Halide/pull/886#discussion_r35996857>:
>
> > @@ -30,7 +31,7 @@ int main(int argc, char **argv) {
> >      // first horizontally, and then vertically.
> >      {
> >          // Take a color 8-bit input
> > -        Image<uint8_t> input = load<uint8_t>("images/rgb.png");
> > +        Image<uint8_t> input = load<Image<uint8_t>>("images/rgb.png");
>
> I guess that would work, but it sure makes my spidey-sense tingle.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/886/files#r35996857>.
>